frontend/model/contracts/group.js

```diff
@@ -579,14 +622,36 @@ sbp('chelonia/defineContract', {
           }
         }
       },
-      process ({ data, meta }, { state, getters }) {
+      process ({ data, meta }, { state }) {
         memberLeaves(state, data.member, meta.createdDate)
       },
-      sideEffect ({ data, contractID }, { state }) {
+      sideEffect ({ data, meta, contractID }, { state }) {
         const rootState = sbp('state/vuex/state')
         const contracts = rootState.contracts || {}
+        const { username } = rootState.loggedIn
+
+        // let user leaves all the chatrooms inside the group
+        if (meta.username === username) {
+          let chatRoomIDsToLeave = []
+          if (!sbp('okTurtles.data/get', 'JOINING_GROUP')) {
+            chatRoomIDsToLeave = Object.keys(state.chatRooms)
+              .filter(cID => rootState[cID] && rootState[cID].users[data.member]) || []
+          } else if (sbp('okTurtles.data/get', 'JOINING_CHATROOM')) {
+            chatRoomIDsToLeave = [state.generalChatRoomId]
+          }
 
-        if (data.member === rootState.loggedIn.username) {
+          sbp('gi.actions/group/leaveChatRooms', {
```

Comment:
Should we `await` here before proceeding?

Reply:
I didn't use `await` here in order to reduce the runtime even a bit.
But I think we can remove `cy.wait` if we use `await` here.
Actually, it won't cause anything wrong in the production mode but now, I am starting to think we should use `await` here.